plugin/githubinator.vim

```diff
@@ -37,19 +37,22 @@ endfunction
 
 " get remote URL from .git/config
 function! s:generate_url() range
-    if !isdirectory('.git')
-        echoerr "githubinator: No .git directory found"
-        return
+    let l:git_root = system('git rev-parse --show-toplevel')
+
+    if v:shell_error != 0
+        echoerr printf('githubinator: %s is NOT managed by git', getcwd())
     endif
 
-    let l:file_name = expand("%")
+    " In order to support relative filepath, convert full path and remove the git root path in it.
+    let l:file_name = expand('%:p')[len(git_root) : -1]
     let l:beg = GetRangeDelimiters()[0]
     let l:end = GetRangeDelimiters()[1]
-    let l:branch = system("git branch 2> /dev/null | awk '{print $2}' | tr -d '\n'")
-    let l:git_remote = system("cat .git/config | grep \"url\" | sed \"s/url.*://g\" | awk '{print \"https://github.com/\", $0}' | tr -d '[:blank:]' | sed \"s/\.git$//g\" | tr -d '\n'")
-    let l:final_url = l:git_remote . '/blob/' . l:branch . '/' . l:file_name . '#L' . l:beg . '-L' . l:end
+    let l:branch = system("git rev-parse HEAD | tr -d '\n'")
```

Comment:
did you mean `git symbolic-ref HEAD` here?

Reply:
oh, I don't know the command.
The answer is No.

It assumes that we work on detached branch.
For instance `git checkout 1ba14847993717796d14d84bbc1852ded124bdde`.
In this case the output of `git branch` is `* (HEAD detached at XXXXXXX)`.
I used `git rev-parse HEAD` in order to support the case :+1:

Reply:
@mopp how are we extracting the actual branch if we are not on a detached branch?

Reply:
@prakashdanish 
We can use the git hash not on a detached branch.

You would like to use branch name not on a detached branch ?

Reply:
I found a nice git command `git name-rev --name-only HEAD` :smile: 

It outputs
* 'master' on master branch.
* 'master~2' on detached from master branch (it means we are on 2 previous commit from master branch)

Github acceps the url like `https://github.com/prakashdanish/vim-githubinator/blob/master~2/doc/githubinator.txt`

Reply:
I noticed this method not works on tagged commit.
For instance, `git name-rev --name-only HEAD` outputs `tags/v1.0.0`.
Then the final URL is `http://github.com/XXX/YYY/blob/tags/v1.0.0/README.ja.md#L6-L6`.
GitHub does not support this URL... 😢

Reply:
I found the URL `http://github.com/XXX/YYY/blob/v1.0.0/README.ja.md#L6-L6`.

A part `tags/` is removed using sed.